jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/VirtualThreadPool.java

```diff
@@ -0,0 +1,210 @@
+import org.eclipse.jetty.util.annotation.ManagedObject;
+import org.eclipse.jetty.util.component.ContainerLifeCycle;
+import org.eclipse.jetty.util.component.Dumpable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link ThreadPool} interface that does not pool, but instead uses {@link VirtualThreads}.
+ */
+@ManagedObject("A thread non-pool for virtual threads")
+public class VirtualThreadPool extends ContainerLifeCycle implements ThreadPool, Dumpable, TryExecutor, VirtualThreads.Configurable
+{
+    private static final Logger LOG = LoggerFactory.getLogger(VirtualThreadPool.class);
+
+    private final AutoLock.WithCondition _joinLock = new AutoLock.WithCondition();
+    private String _name = null;
+    private Executor _virtualExecutor;
+    private Thread _main;
+    private boolean _tracking;
+    private boolean _detailedDump;
+
+    public VirtualThreadPool()
+    {
+        if (!VirtualThreads.areSupported())
+            throw new IllegalStateException("Virtual Threads not supported");
+    }
+
+    /**
+     * @return the name of this thread pool
+     */
+    @ManagedAttribute("name of the thread pool")
+    public String getName()
+    {
+        return _name;
+    }
+
+    /**
+     * <p>Sets the name of this thread pool, used as a prefix for the thread names.</p>
+     *
+     * @param name the name of this thread pool
+     */
+    public void setName(String name)
+    {
+        if (isRunning())
+            throw new IllegalStateException(getState());
+        if (StringUtil.isBlank(name) && name != null)
+            throw new IllegalArgumentException("Blank name");
+        _name = name;
+    }
+
+    public void setTracking(boolean tracking)
+    {
+        if (isRunning())
+            throw new IllegalStateException(getState());
+        _tracking = tracking;
+    }
+
+    public boolean isTracking()
+    {
+        return _tracking;
+    }
+
+    @ManagedAttribute("reports additional details in the dump")
+    public boolean isDetailedDump()
+    {
+        return _detailedDump;
+    }
+
+    public void setDetailedDump(boolean detailedDump)
+    {
+        _detailedDump = detailedDump;
+        if (_virtualExecutor instanceof TrackingExecutor trackingExecutor)
+            trackingExecutor.setDetailedDump(detailedDump);
+    }
+    
+    @Override
+    protected void doStart() throws Exception
+    {
+        _main = new Thread("virtual main")
+        {
+            @Override
+            public void run()
+            {
+                try (AutoLock.WithCondition l = _joinLock.lock())
+                {
+                    while (isRunning())
+                    {
+                        l.await();
+                    }
+                }
+                catch (InterruptedException e)
+                {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        _main.start();
```

Comment:
Is this platform thread only necessary to keep the server alive?
The thread name should not contain "virtual" because it's not virtual. How about "keepalive"?